packages/common/random-access-storage/src/browser/web-fs.ts

```diff
@@ -323,51 +329,50 @@ export class WebFile extends EventEmitter implements File {
   }
 
   private async _flushNow() {
-    this._flushPromise = (this._flushPromise ?? Promise.resolve())
-      .then(() => this._flushCache())
-      .catch((err) => log.warn(err));
+    this._flushPromise = this._flushPromise ?? Promise.resolve();
+
+    try {
+      await this._flushCache();
+      this._flushPromise = null;
```

Comment:
the logic in scheduled flush was broken:
1. _flushPromise was initialized.
2. callback captures and awaits it.
3. then compares whether promise changed to avoid double-flush, but it was the same

Reply:
This doesnt look right: `_flushNow` must also guard against concurrent invocations of `_flushCache`. I think here two concurrent calls to `_flushNow` will also result in `_flushCache` running concurrently too

Reply:
oh true, the flow is super weird after the change, but as for the comment:

>  I think here two concurrent calls to _flushNow will also result in _flushCache running concurrently too

I think that's the desired behavior in the realm of weird api usages.
imagine I write and flush while the ongoing flush went past its write call. In this case there'll be unflushed data in our buffer.
it's probably best to introduce a `dirty` flag here and decide whether to start a new flush based on it.